cdap-ui/app/hydrator/controllers/create/partials/pipelineupgrade-modal-ctrl.js

```diff
@@ -147,7 +148,7 @@ angular.module(PKG.name + '.feature.hydrator')
       newConfig.config.stages = stages;
       newConfig.config.postActions = postActions;
 
-      if (newConfig.__ui__) {
+      if (newConfig.__ui__ && this.isImport) {
```

Comment:
why do we only want to delete this on import?

Reply:
We used to delete this every time we import and fix something. This was causing the draft that's already saved to change its id and causing it to save as a new draft. For upgrading a draft, we already have an id and do not need to delete the id. So, here we're excluding deletion of this property if it's an upgrade of a draft.

Reply:
Changed the way we preserve the draftId - instead of not deleting `__ui__` at all for upgrades, we now add any existing draftId back after we clear `__ui__`